src/Frontend/src/components/messages2/FlowDiagram.vue

```diff
@@ -166,16 +169,21 @@ onMounted(async () => {
 
   const assignDescendantLevelsAndWidth = (message: MappedMessage, level = 0) => {
     message.level = level;
-    const children = mappedMessages.filter((mm) => mm.parentId === message.messageId && mm.parentEndpoint === message.receivingEndpoint);
+    const children = mappedMessages.filter((mm) => mm.parentId === message.messageId && mm.parentEndpoint === message.receivingEndpoint.name);
     message.width =
       children.length === 0
         ? 1 //leaf node
-        : children.map((child) => (child.width == null ? assignDescendantLevelsAndWidth(child, level + 1) : child)).reduce((sum, { width }) => sum + width!, 0);
+        : children.map((child) => (child.width === 0 ? assignDescendantLevelsAndWidth(child, level + 1) : child)).reduce((sum, { width }) => sum + width, 0);
     return message;
   };
-  for (const root of mappedMessages.filter((message) => !message.parentId)) assignDescendantLevelsAndWidth(root);
+  for (const root of mappedMessages.filter((message) => !message.parentId)) {
+    assignDescendantLevelsAndWidth(root);
+  }
+
+  const nodes = constructNodes(mappedMessages);
+  const edges = constructEdges(nodes.map((n) => n.data as MappedMessage));
```

Comment:
only passing the result of constructNodes to make up the edges, this prevents edges without parents